### ISSUE TITLE
noobaa install hanging with system connecting

### ISSUE THREAD

**Issue by reesede** [noobaa/noobaa-operator#696]
I'm trying to install Noobaa on an OpenShift cluster and the "noobaa" install command hangs with the message:

     INFO[0030] ⏳ System Phase is "Connecting". Waiting for phase ready ...

The output from "noobaa install" is:
==========================
noobaa install
INFO[0000] CLI version: 5.7.0                           
INFO[0000] noobaa-image: noobaa/noobaa-core:5.7.0       
INFO[0000] operator-image: noobaa/noobaa-operator:5.7.0 
INFO[0000] Namespace: noobaa                            
INFO[0000]                                              
INFO[0000] CRD Create:                                  
INFO[0001] ✅ Already Exists: CustomResourceDefinition "noobaas.noobaa.io" 
INFO[0001] ✅ Already Exists: CustomResourceDefinition "backingstores.noobaa.io" 
INFO[0001] ✅ Already Exists: CustomResourceDefinition "namespacestores.noobaa.io" 
INFO[0001] ✅ Already Exists: CustomResourceDefinition "bucketclasses.noobaa.io" 
INFO[0001] ✅ Already Exists: CustomResourceDefinition "objectbucketclaims.objectbucket.io" 
INFO[0001] ✅ Already Exists: CustomResourceDefinition "objectbuckets.objectbucket.io" 
INFO[0001]                                              
INFO[0001] Operator Install:                            
INFO[0002] ✅ Already Exists: Namespace "noobaa"         
INFO[0002] ✅ Created: ServiceAccount "noobaa"           
INFO[0002] ✅ Created: Role "noobaa"                     
INFO[0002] ✅ Created: RoleBinding "noobaa"              
INFO[0002] ✅ Created: ClusterRole "noobaa.noobaa.io"    
INFO[0002] ✅ Created: ClusterRoleBinding "noobaa.noobaa.io" 
INFO[0002] ✅ Created: Deployment "noobaa-operator"      
INFO[0002]                                              
INFO[0002] System Create:                               
INFO[0002] ✅ Already Exists: Namespace "noobaa"         
INFO[0002] ✅ Created: NooBaa "noobaa"                   
INFO[0002]                                              
INFO[0002] NOTE:                                        
INFO[0002]   - This command has finished applying changes to the cluster. 
INFO[0002]   - From now on, it only loops and reads the status, to monitor the operator work. 
INFO[0002]   - You may Ctrl-C at any time to stop the loop and watch it manually. 
INFO[0002]                                              
INFO[0002] System Wait Ready:                           
INFO[0003] ⏳ System Phase is "". Pod "noobaa-operator-fcb746986-zkdx7" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [noobaa-operator]). ContainersNotReady (containers with unready status: [noobaa-operator]).  
INFO[0006] ⏳ System Phase is "". Pod "noobaa-operator-fcb746986-zkdx7" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [noobaa-operator]). ContainersNotReady (containers with unready status: [noobaa-operator]).  
INFO[0009] ⏳ System Phase is "". Pod "noobaa-operator-fcb746986-zkdx7" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [noobaa-operator]). ContainersNotReady (containers with unready status: [noobaa-operator]).  
INFO[0012] ⏳ System Phase is "". StatefulSet "noobaa-core" is not found yet 
INFO[0015] ⏳ System Phase is "". Pod "noobaa-core-0" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [core]). ContainersNotReady (containers with unready status: [core]).  
INFO[0018] ⏳ System Phase is "Connecting". Pod "noobaa-core-0" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [core]). ContainersNotReady (containers with unready status: [core]).  
INFO[0021] ⏳ System Phase is "Connecting". Pod "noobaa-core-0" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [core]). ContainersNotReady (containers with unready status: [core]).  
INFO[0024] ⏳ System Phase is "Connecting". Pod "noobaa-core-0" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [core]). ContainersNotReady (containers with unready status: [core]).  
INFO[0027] ⏳ System Phase is "Connecting". Pod "noobaa-core-0" is not yet ready: Phase="Pending". ContainersNotReady (containers with unready status: [core]). ContainersNotReady (containers with unready status: [core]).  
INFO[0030] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0033] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0036] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0039] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0042] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0045] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0048] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0051] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0054] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0057] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0060] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0063] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0066] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0069] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0072] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0075] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0078] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0081] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0084] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0087] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0090] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0093] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
INFO[0096] ⏳ System Phase is "Connecting". Waiting for phase ready ... 

Output from "noobaa status":
======================
noobaa status
INFO[0001] CLI version: 5.7.0                           
INFO[0001] noobaa-image: noobaa/noobaa-core:5.7.0       
INFO[0001] operator-image: noobaa/noobaa-operator:5.7.0 
INFO[0001] noobaa-db-image: centos/mongodb-36-centos7   
INFO[0001] Namespace: noobaa                            
INFO[0001]                                              
INFO[0001] CRD Status:                                  
INFO[0001] ✅ Exists: CustomResourceDefinition "noobaas.noobaa.io" 
INFO[0001] ✅ Exists: CustomResourceDefinition "backingstores.noobaa.io" 
INFO[0001] ✅ Exists: CustomResourceDefinition "namespacestores.noobaa.io" 
INFO[0001] ✅ Exists: CustomResourceDefinition "bucketclasses.noobaa.io" 
INFO[0001] ✅ Exists: CustomResourceDefinition "objectbucketclaims.objectbucket.io" 
INFO[0001] ✅ Exists: CustomResourceDefinition "objectbuckets.objectbucket.io" 
INFO[0001]                                              
INFO[0001] Operator Status:                             
INFO[0001] ✅ Exists: Namespace "noobaa"                 
INFO[0001] ✅ Exists: ServiceAccount "noobaa"            
INFO[0001] ✅ Exists: Role "noobaa"                      
INFO[0001] ✅ Exists: RoleBinding "noobaa"               
INFO[0001] ✅ Exists: ClusterRole "noobaa.noobaa.io"     
INFO[0001] ✅ Exists: ClusterRoleBinding "noobaa.noobaa.io" 
INFO[0001] ✅ Exists: Deployment "noobaa-operator"       
INFO[0001]                                              
INFO[0001] System Status:                               
INFO[0002] ✅ Exists: NooBaa "noobaa"                    
INFO[0002] ✅ Exists: StatefulSet "noobaa-core"          
INFO[0002] ✅ Exists: Service "noobaa-mgmt"              
INFO[0002] ✅ Exists: Service "s3"                       
INFO[0002] ✅ Exists: StatefulSet "noobaa-db"            
INFO[0002] ✅ Exists: Service "noobaa-db"                
INFO[0002] ✅ Exists: Secret "noobaa-server"             
INFO[0002] ❌ Not Found: Secret "noobaa-operator"        
INFO[0002] ❌ Not Found: Secret "noobaa-endpoints"       
INFO[0002] ❌ Not Found: Secret "noobaa-admin"           
INFO[0002] ✅ Exists: Secret "noobaa-root-master-key"    
INFO[0002] ❌ Not Found: StorageClass "noobaa.noobaa.io" 
INFO[0002] ❌ Not Found: BucketClass "noobaa-default-bucket-class" 
INFO[0002] ❌ Not Found: Deployment "noobaa-endpoint"    
INFO[0002] ❌ Not Found: HorizontalPodAutoscaler "noobaa-endpoint" 
INFO[0002] ⬛ (Optional) Not Found: BackingStore "noobaa-default-backing-store" 
INFO[0002] ⬛ (Optional) Not Found: CredentialsRequest "noobaa-aws-cloud-creds" 
INFO[0002] ⬛ (Optional) Not Found: CredentialsRequest "noobaa-azure-cloud-creds" 
INFO[0002] ⬛ (Optional) Not Found: Secret "noobaa-azure-container-creds" 
INFO[0002] ⬛ (Optional) Not Found: Secret "noobaa-gcp-bucket-creds" 
INFO[0002] ⬛ (Optional) Not Found: CredentialsRequest "noobaa-gcp-cloud-creds" 
INFO[0002] ⬛ (Optional) Not Found: PrometheusRule "noobaa-prometheus-rules" 
INFO[0002] ⬛ (Optional) Not Found: ServiceMonitor "noobaa-mgmt-service-monitor" 
INFO[0002] ⬛ (Optional) Not Found: ServiceMonitor "s3-service-monitor" 
INFO[0002] ✅ (Optional) Exists: Route "noobaa-mgmt"     
INFO[0002] ✅ (Optional) Exists: Route "s3"              
INFO[0002] ✅ Exists: PersistentVolumeClaim "db-noobaa-db-0" 
INFO[0002] ❌ System Phase is "Connecting"               
INFO[0003] ⏳ System Phase is "Connecting". Waiting for phase ready ... 
#------------------#
#- Backing Stores -#
#------------------#

No backing stores found.

#------------------#
#- Bucket Classes -#
#------------------#

No bucket classes found.

#-----------------#
#- Bucket Claims -#
#-----------------#

No OBCs found.

Output from "noobaa diagnose"
=======================
noobaa diagnose
INFO[0000] Running collection of diagnostics            
INFO[0001] ❌ No noobaa-s3 pods found\n                  
INFO[0001] ❌ No noobaa-db pods found\n                  

Output from "noobaa version"
======================
noobaa version
INFO[0000] CLI version: 5.7.0                           
INFO[0000] noobaa-image: noobaa/noobaa-core:5.7.0       
INFO[0000] operator-image: noobaa/noobaa-operator:5.7.0 

Output from "oc version"
==================
oc version
Client Version: 4.6.0-202103202154.p0-c3dc785
Server Version: 4.6.23
Kubernetes Version: v1.19.0+263ee0d

I'm very new to Noobaa, so any help would be appreciated.

Thanks!

**Comment by guymguym** [comment]
Hi @reesede Thank you for opening the issue!

Lets try to help with troubleshooting, and figure out  how to improve our status on the way.

So when the operator reports it's on Connecting phase, it means it is trying to connect to the noobaa-core and noobaa-db pods to configure the system. If this is stuck like this for more than a few minutes, there is a good chance that these pods are either not running because kubernetes is waiting for resource allocation such as CPU/MEM or PV. Can you start by check the status of those like this in the namespace where you deploy noobaa?
```
kubectl get pod
kubectl get sts
kubectl get pvc
```

I think that our status checks in the operator status have not been adjusted to detect problems in the noobaa-db pod, so it only reports issues with noobaa-core pod. CC @nimrod-becker .

**Comment by guymguym** [comment]
This CheckWaitingFor function is the one where we try to report what the operator is waiting for, and seems like it just doens't check the DB pod/PVC yet - which is a gap in troubleshooting that we can close -

https://github.com/noobaa/noobaa-operator/blob/93acc35537cf21cf1c49afd86ca319cf55bdecc2/pkg/system/system.go#L662-L664

**Comment by reesede** [comment]
kubectl get pod
NAME                              READY   STATUS    RESTARTS   AGE
noobaa-core-0                     1/1     Running   0          98s
noobaa-operator-fcb746986-cv56g   1/1     Running   0          104s



kubectl get sts
NAME          READY   AGE
noobaa-core   1/1     2m10s
noobaa-db     0/1     2m10s



kubectl get pvc
NAME             STATUS    VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   AGE
db-noobaa-db-0   Pending                                                     2m42s



oc describe pvc db-noobaa-db-0
Name:          db-noobaa-db-0
Namespace:     noobaa
StorageClass:  
Status:        Pending
Volume:        
Labels:        app=noobaa
               noobaa-db=noobaa
Annotations:   <none>
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Mounted By:    <none>
Events:
  Type    Reason         Age                  From                         Message
  ----    ------         ----                 ----                         -------
  Normal  FailedBinding  8s (x14 over 3m10s)  persistentvolume-controller  no persistent volumes available for this claim and no storage class is set




> Hi @reesede Thank you for opening the issue!
> 
> Lets try to help with troubleshooting, and figure out how to improve our status on the way.
> 
> So when the operator reports it's on Connecting phase, it means it is trying to connect to the noobaa-core and noobaa-db pods to configure the system. If this is stuck like this for more than a few minutes, there is a good chance that these pods are either not running because kubernetes is waiting for resource allocation such as CPU/MEM or PV. Can you start by check the status of those like this in the namespace where you deploy noobaa?
> 
> ```
> kubectl get pod
> kubectl get sts
> kubectl get pvc
> ```
> 
> I think that our status checks in the operator status have not been adjusted to detect problems in the noobaa-db pod, so it only reports issues with noobaa-core pod. CC @nimrod-becker .

**Comment by guymguym** [comment]
@reesede So the noobaa-db PVC is not getting provisioned. 
Do you have a default storage class?
If you cannot set a default class, you can also use the noobaa cli to manually select a class for the DB `--db-storage-class` and for the PVPool `--pv-pool-default-storage-class`.
Check out the `noobaa options` for full list of flags.

**Comment by guymguym** [comment]
See here - https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/

**Comment by reesede** [comment]
Thanks. I did finally get past this problem. There appear to have been several issues. 

First was the storageClass (above). 

I also ran into the issue at [https://github.com/noobaa/noobaa-operator/issues/539](https://github.com/noobaa/noobaa-operator/issues/539). 

Finally, the db-noobaa-db-0 required a VolumeMode of Filesystem. My system used block storage, and I had to change one of my drives to Filesystem.
